src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -40,15 +40,9 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
                     // singleFrameOnly check first so to prevent frame superceding when disabled.
                     if (!data.singleFrameOnly === false || e.sender.isValidWithFrameConnect(e.frameRoutingId)) {
                         Promise.resolve()
-                            .then(() => endpoint.apiFunc(identity, data, ack, nack))
-                            .then(result => {
-                                // older action calls will invoke ack internally, newer ones will return a value
-                                if (result !== undefined) {
-                                    ack(new AckPayload(result));
-                                }
-                            }).catch(err => {
-                                nack(err);
-                            });
+                            .then(() => endpoint.apiFunc(identity, data))
+                            .then(result => { ack(new AckPayload(result)); })
+                            .catch(err => { nack(err); });
```

Comment:
`catch(nack)`

Reply:
will change it.
...although technically, that would propagate some unpredictable `error` value (whatever nack returns)... on the other hand the present code doesn't propagate either (returns `undefined`). but it doesn't matter: in any case it's the end of the chain anyway and the promise goes away at that point. but should a future maintainer add another catch, they would need to make sure this one returns its input.